lib/gltfPipeline.js

```diff
@@ -0,0 +1,130 @@
+'use strict';
+var fs = require('fs');
+var addDefaults = require('./addDefaults');
+var removeUnusedImages = require('./removeUnusedImages');
+var removeUnusedSamplers = require('./removeUnusedSamplers');
+var removeUnusedShaders = require('./removeUnusedShaders');
+var removeUnusedTechniques = require('./removeUnusedTechniques');
+var removeUnusedPrograms = require('./removeUnusedPrograms');
+var removeUnusedBuffers = require('./removeUnusedBuffers');
+var removeUnusedBufferViews = require('./removeUnusedBufferViews');
+var removeUnusedMaterials = require('./removeUnusedMaterials');
+var removeUnusedSkins = require('./removeUnusedSkins');
+var removeUnusedCameras = require('./removeUnusedCameras');
+var removeUnusedTextures = require('./removeUnusedTextures');
+var removeUnusedMeshes = require('./removeUnusedMeshes');
+var removeUnusedNodes = require('./removeUnusedNodes');
+var removeUnusedAccessors = require('./removeUnusedAccessors');
+var removeUnused = require('./removeUnused');
+var parseBinaryGltf = require('./parseBinaryGltf');
+var addPipelineExtras = require('./addPipelineExtras');
+var convertDagToTree = require('./convertDagToTree');
+var combineMeshes = require('./combineMeshes');
+var combinePrimitives = require('./combinePrimitives');
+var removeUnusedVertices = require('./removeUnusedVertices');
+var OptimizationStatistics = require('./OptimizationStatistics');
+var writeGltf = require('./writeGltf');
+var writeBinaryGltf = require('./writeBinaryGltf');
+var readGltf = require('./readGltf');
+var removePipelineExtras = require('./removePipelineExtras');
+var loadGltfUris = require('./loadGltfUris');
+var Cesium = require('cesium');
+var defaultValue = Cesium.defaultValue;
+var defined = Cesium.defined;
+
+module.exports = {
+    processJSON : processJSON,
+    processJSONWithExtras : processJSONWithExtras,
+    processFile : processFile,
+    processJSONToDisk : processJSONToDisk,
+    processFileToDisk : processFileToDisk,
+    writeFile : writeFile
+};
```

Comment:
If possible we should try not to expose `processJSONWithExtras` and `writeFile` since they are really just internal to this file. Some tests may have to adjust for this.